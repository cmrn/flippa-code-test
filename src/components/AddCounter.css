.AddCounter {
  margin: 20px 0;
  display: flex;
  font-size: 20px;
}


.AddCounter-title, 
.AddCounter-button {
  border: 1px solid grey;
}

.AddCounter-title {
  flex: 1;
  min-width: 0;
  border-right: none;
  padding: 10px;
}

.AddCounter-button:hover {
  cursor: pointer;
}