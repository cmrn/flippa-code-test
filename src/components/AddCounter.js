import React from 'react';
import PropTypes from 'prop-types';
import cnames from 'classnames';
import './AddCounter.css';

class AddCounter extends React.Component {
  constructor(props) {
    super(props);
    this.state = { title: '' };
    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  };

  handleTitleChange(event) {
    this.setState({ title: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.onAdd(this.state.title);
    this.setState({ title: '' });
  }

  render() {
    const { className, onAdd, ...props} = this.props;
    return (
      <form {...props} className={cnames("AddCounter", className)} onSubmit={this.handleSubmit}>
        <input className="AddCounter-title" type="text" value={this.state.title} onChange={this.handleTitleChange} />
        <input className="AddCounter-button" type="submit" value="Add Counter" />
      </form>
    );
  }
};

AddCounter.propTypes = {
  onAdd: PropTypes.func.isRequired,
  className: PropTypes.string,
};

export default AddCounter;
