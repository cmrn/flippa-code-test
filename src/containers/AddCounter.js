import React from 'react';
import AddCounter from '../components/AddCounter';
import { view } from 'react-easy-state';
import counterStore from '../counterStore';

const AddCounterContainer = (props) => (
  <AddCounter {...props} onAdd={counterStore.add} />
);

export default view(AddCounterContainer);
