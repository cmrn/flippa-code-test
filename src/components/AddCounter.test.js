import React from 'react';
import { shallow } from 'enzyme';
import AddCounter from './AddCounter';

it('renders a text box and button', () => {
  const wrapper = shallow(<AddCounter onAdd={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

describe('when the form is submitted', () => {
  it('calls the onAdd handler', () => {
    const handler = jest.fn();
    const wrapper = shallow(<AddCounter onAdd={handler} />);
    wrapper.find('.AddCounter').simulate('submit', { preventDefault: ()=>{} });
    expect(handler.mock.calls.length).toBe(1);
  });

  it('uses the title from the input', () => {
    const title = 'foo';
    const handler = jest.fn();
    const wrapper = shallow(<AddCounter onAdd={handler} />);
    wrapper.find('.AddCounter-title').simulate('change', { target: { value: title }});
    wrapper.find('.AddCounter').simulate('submit', { preventDefault: ()=>{} });
    expect(handler.mock.calls[0][0]).toEqual(title);
  });
});
