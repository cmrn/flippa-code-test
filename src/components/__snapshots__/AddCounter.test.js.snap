// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a text box and button 1`] = `
<form
  className="AddCounter"
  onAdd={[Function]}
  onSubmit={[Function]}
>
  <input
    className="AddCounter-title"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    className="AddCounter-button"
    type="submit"
    value="Add Counter"
  />
</form>
`;
